### PR TITLE
chore: prepare BF event

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
Closes #.

### Summary

Add functions for sales events like Black Friday.

### Screenshots 

image

image

----

### Test instructions

1. Check if the banners appear with the correct link/URL. More about this here: https://github.com/Codeinwp/themeisle/issues/1575#issuecomment-1757321452
2. By using all three plugins (Neve, Otter, Feedzy), check the admin notice (the small one) priority. There should be notice only for the highest priority product. The order is Neve > Otter > Feedzy. More on this here: https://github.com/Codeinwp/themeisle/issues/1575#issuecomment-1761388729

The giant banners will have a dynamic text that shows the remaining time in the top-left corner. The start is set early to facilitate the test (it will be changed after the approval). ~The time interval set for testing is 2023-10-10 00:00:00 and 2023-11-27 23:59:00.~ If possible, try to change the server time to see if the banner is working correctly when the time is over. 

The PRs are:
- Neve: https://github.com/Codeinwp/neve/pull/4121
- Otter: https://github.com/Codeinwp/otter-blocks/pull/1922
- Feedzy: https://github.com/Codeinwp/feedzy-rss-feeds/pull/805

---- 

### Checklist before the final review

- [ ] Included E2E or unit tests for the changes in this PR.
- [x] Visual elements are not affected by independent changes.
- [x] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [x] It loads additional script in frontend only if it is required.
- [x] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [x] In case of deprecation, old blocks are safely migrated.
- [x] It is usable in Widgets and FSE.
- [x] Copy/Paste is working if the attributes are modified.
- [x] PR is following [the best practices]()

### CODE REVIEW

File: inc/plugins/class-limited-offers.php [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+	 * @param string $slug Slug of the deal.
+	 * @return string Remaining time for the deal.
+	 */
+	public function get_remaining_time_for_deal( $slug ) {
+		if ( empty( $slug ) ) {
+			return '';
+		}
+
+		if ( 'bf' === $slug ) {
+			$end_date = '2023-11-27 23:59:00';
```
Comment (HardeepAsrani) [inline inc/plugins/class-limited-offers.php:135]: We can set the start, end date is a constant on top so we don't have to reference it in multiple places like this.

**Review by HardeepAsrani — APPROVED** [review]
Left one comment but it looks good. We can get all of them in QA together to confirm they also work fine together. And we can also make the product specific changes to define priority properly.

File: src/dashboard/style.scss [hunk 2]
```diff
@@ -1,5 +1,7 @@
 @import "../blocks/plugins/feedback/editor.scss";
 
+@import url('https://fonts.cdnfonts.com/css/barlow');
```
Comment (HardeepAsrani) [inline src/dashboard/style.scss:3]: I agree with Cristi here, we can remove this and see if we can try for a similar style without loading an extra font.

**Comment by HardeepAsrani** [comment]
Also as talked on Slack, we can show the notice on all screens on Otter's main dashboard.

File: src/dashboard/components/Deal.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/**
+ * WordPress dependencies.
+ */
+import { __ } from '@wordpress/i18n';
+
+const Deal = ( props ) => {
+	return (
+		<div className="otter-deal">
+			<a href={props.link}>
```
Comment (HardeepAsrani) [inline src/dashboard/components/Deal.js:9]: All looks good, we can just make sure the banner opens in a new tab. @Soare-Robert-Daniel You can make the small change and put it for QA & ping Harshit to take a look.

**Review by HardeepAsrani — APPROVED** [review]
~One more issue, I found is that the notices appear even if you've Pro version installed. As mentioned in the issue, the notice shouldn't be visible if the user has Pro version.~

Sorry, hit approved by mistake.

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
One more issue, I found is that the notices appear even if you've Pro version installed. As mentioned in the issue, the notice shouldn't be visible if the user has Pro version.

**Comment by harshitarora-in** [comment]
@Soare-Robert-Daniel Small fix is needed in the LTD banner link, it should be: https://themeisle.com/plugins/otter-blocks/blackfriday?utm_source=wpadmin&utm_medium=dashboard&utm_campaign=blackfridayltd23&utm_content=otter-blocksblackfriday

In the current version, you have used utm_medium=globalnotice but it should be utm_medium=dashboard

**Comment by harshitarora-in** [comment]
@Soare-Robert-Daniel @HardeepAsrani If I dismissed the Neve LTD notice when will the Otter Blocks dismissible notice get triggered?

**Comment by HardeepAsrani** [comment]
@harshitarora-in Do we want to display notices back to back? We've avoided showing notices to users once they dismiss one. Based on past experience, users don't seem very happy if we show them too many notices within a small amount of time. Let me know what you think.

**Comment by harshitarora-in** [comment]
@HardeepAsrani Yes Yes, I agree. I'm just asking about the current delay between two dismissible notices.

**Comment by HardeepAsrani** [comment]
@harshitarora-in Right now we don't display another Global notice if one has been dismissed.

**Comment by harshitarora-in** [comment]
> @harshitarora-in Right now we don't display another Global notice if one has been dismissed.

Got it, thanks for the clarification.

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
@Soare-Robert-Daniel The banner still appears when you have Otter Pro.

Screenshot 2023-10-19 at 6 13 27 PM

**Comment by Soare-Robert-Daniel** [comment]
As a side note here for future things, the pro verification must made at least in `init` hook. The direct verification works only on dev local mode since all otter files are loaded all at once.

**Comment by rodica-andronache** [comment]
@Soare-Robert-Daniel all good now, thank you!

**Pull request merged**